enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,845 @@
+   stanza is patched with the architecture node selector requirements. If any
+   node selector term is present, they are all patched. Otherwise, a new node
+   selector term is added to the `.matchExpressions` list. The merge strategy
+   described here follows the rules in KEP-3838[^2]
+7. The scheduling gate is removed from the pod spec and the pod is updated
+8. The scheduler notices that no scheduling gates are set and continues the
+   scheduling process for the given pod
+
+The above two main workflows are depicted in the following sequence diagram:
+
+```mermaid
+%%{init: {'securityLevel': 'loose', 'theme':'neutral', 'themeVariables': { 'fontSize':'14px', 'fontFamily':'Roboto Mono'}}}%%
+sequenceDiagram
+    actor User/Controller
+    participant APIServer
+    participant MutatingAdmissionWebhook
+    participant Pod
+    participant Scheduler
+    participant PodPlacementController
+    participant Image Registry
+
+    activate APIServer
+    User/Controller ->> APIServer: Submit Pod
+    APIServer ->> MutatingAdmissionWebhook: request(CREATE, pod)
+    MutatingAdmissionWebhook ->> Pod: Add Scheduling Gate
+    MutatingAdmissionWebhook ->> APIServer: patched pod
+    deactivate APIServer
+
+    par
+        APIServer -->> Scheduler: Notify pod event
+        alt len(schedulingGates) > 0
+        Scheduler -->> Scheduler: hold in the queue until no scheduling gates are set
+        end
+
+    and
+        APIServer -->> PodPlacementController: Notify pod event
+
+        alt schedulingGatePresent
+            activate PodPlacementController
+            PodPlacementController ->> Image Registry: Inspect Container Images
+
+            Image Registry -->> PodPlacementController: supported architectures
+            PodPlacementController ->> PodPlacementController: Intersect Supported Architectures
+            PodPlacementController ->> Pod: PatchNodeAffinity
+            Note over PodPlacementController: inspection errors are ignored<br>to allow the removal of the scheduling<br>gate.
+            deactivate PodPlacementController
+            PodPlacementController ->> Pod: RemoveSchedulingGate
+        end
+    end
+    Note over Scheduler: no more schedulingGates
+    Scheduler ->> Scheduler: move Pod to activeQ <br> and other plugins
+
+```
+#### Deletion of the CR
+1. The cluster admin deletes the CR
+2. All resources related to the operator - the controllers, webhook service and
+   pods automatically get deleted
+3. Any pods stuck in the scheduling gated phase will be patched without changes to
+   the node affinity and their scheduling will ungate
+4. Recreating the CR again will install all related resources
+
+#### Operator uninstallation
+- Operator uninstallation removes the operator. If the option to uninstall
+  operands is selected, all resources are removed.
+- Any pods stuck in the scheduling gated phase could need a manual patch
+  performed by the users to remove the scheduling gate and resume scheduling
```

Comment:
added this to the new documentation plan section.

Reply:
above it says deleting the CR will delete the controllers, so how does a use "remove the CR, then wait until scheduling gates are removed before uninstalling the operator"?

uninstalling the operator actually has no effect on the controller(operand) anyway, unless you choose delete operands, in which case that's the same as deleting the CR (where, again, it's not clear how a user can choose to wait before deleting the controller).

Reply:
There are a few cases in terms of deletion of the operator:
- uninstall operator
-  uninstall operator + operands - pods might be left in scheduling gated state which would need to be manually patched . We noted this in the risks/documentation section where we would need to be explicit and provide clear guidance to the user on this case.
- delete CR + uninstall operator - this is essentially the same as the above case, but deleting the CR first would trigger the controller to start patching the pods which are scheduling gated manually. In case the user prematurely removes the operator (by also selecting remove operands), pods will be stuck in scheduling gated state which would have to be removed manually.

Is it also possible to not expose the option for a particular operator to remove operands. i.e, not make that check box available? in that case the uninstall option would only remove the operator, but the cleanup of operand resources would have to be manual if the operator is removed before the CR.

Reply:
In any case, the plan for now is to allow uninstallation of the operator (with operands) in which case the pods which are scheduling gated will have to be patched manually - this will be a documentation effort. The other method is deletion of the CR which will un-gate any scheduling gated pods provided the user gives sufficient time by monitoring these pods to see if the gate has been removed. These points have been captured in this proposal.

Reply:
> Is it also possible to not expose the option for a particular operator to remove operands. i.e, not make that check box available?

This is actually possible. See https://github.com/openshift/console/pull/12234

You would put the following annotation in the CSV:
```yaml
"console.openshift.io/disable-operand-delete": "true"
```

Hopefully the annotation makes clear that this is only something that console takes action on. Other clients (e.g. oc, kubectl) and OLM do not do anything to prevent operand deletion coinciding with the operator uninstall.

Reply:
> Hopefully the annotation makes clear that this is only something that console takes action on. Other clients (e.g. oc, kubectl) and OLM do not do anything to prevent operand deletion coinciding with the operator uninstall.

but they(oc, kubectl, other clients besides console) also don't implement anything to delete(or allow the user to request deletion of) the operands during operator uninstall, right?  Not that someone couldn't create a client that does, of course.

Reply:
Correct. Console has some special logic (via some vendored `kubectl-operator` plugin code) to list the operands for a particular operator, and then perform a semi-orderly deletion.

Other clients are on their own to deal with deleting operands, CSVs, subscriptions, RBAC, etc.

Reply:
Having this option disabled will prevent the user from deleting operands when the scheduling gates are being removed (triggered through CR deletion). Of course, the user could always force delete through oc, but coupled with good documentation - this can be mitigated.